src/code/z_camera.c

```diff
@@ -543,323 +868,7415 @@ s32 Camera_GetWaterBoxBgCamSetting(Camera* camera, f32* waterY) {
+    offsetForwards.x = focalActorPosRot.pos.x + (30.0f * forwardsUnitX);
+    offsetForwards.y = focalActorPosRot.pos.y + focalActorHeight;
+    offsetForwards.z = focalActorPosRot.pos.z + (30.0f * forwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal1.s")
+    camCollision.pos.x = focalActorPosRot.pos.x + (12.0f * forwardsUnitX);
+    camCollision.pos.y = offsetForwards.y;
+    camCollision.pos.z = focalActorPosRot.pos.z + (12.0f * forwardsUnitZ);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal2.s")
+    if ((camera->play->state.frames % 2) != 0) {
+        // Turn and project 65.92 degrees left
+        yawForwardsOffset = focalActorPosRot.rot.y + DEG_TO_BINANG(65.92f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal3.s")
+        angledOffsetForwards.x = (Math_SinS(yawForwardsOffset) * 50.0f) + offsetForwards.x;
+        angledOffsetForwards.y = offsetForwards.y;
+        angledOffsetForwards.z = (Math_CosS(yawForwardsOffset) * 50.0f) + offsetForwards.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal4.s")
+        if ((Camera_BgCheckInfo(camera, &angledOffsetForwards, &camCollision) != 0) &&
+            Camera_CheckOOB(camera, &offsetForwards, &angledOffsetForwards)) {
+            // Always going to result in 28.0f?
+            distResult = OLib_Vec3fDistXZ(&offsetForwards, &camCollision.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Normal0.s")
+            // (-D_801B9E60 < (50.0f - distResult))
+            if (!((D_801B9E5C == 2) && (-D_801B9E60 >= (50.0f - distResult)))) {
+                D_801B9E5C = 1;
+                distResult = 50.0f - distResult; // 22.0f
+                D_801B9E60 = distResult;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Parallel1.s")
+                // if (distResult == 0.0f)
+                if (distResult == -distResult) {
+                    distResultAdj = 0.0f;
+                } else {
+                    distResultAdj = distResult;
+                }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Parallel2.s")
+                return distResultAdj;
+            }
+        }
+    } else {
+        // Turn and project 90 degrees right
+        yawForwardsOffset = focalActorPosRot.rot.y - DEG_TO_BINANG(90);
+
+        angledOffsetForwards.x = (Math_SinS(yawForwardsOffset) * 50.0f) + offsetForwards.x;
+        angledOffsetForwards.y = offsetForwards.y;
+        angledOffsetForwards.z = (Math_CosS(yawForwardsOffset) * 50.0f) + offsetForwards.z;
+
+        if ((Camera_BgCheckInfo(camera, &angledOffsetForwards, &camCollision) != 0) &&
+            Camera_CheckOOB(camera, &offsetForwards, &angledOffsetForwards)) {
+            // Always going to result in 28.0f?
+            distResult = OLib_Vec3fDistXZ(&offsetForwards, &camCollision.pos);
+
+            if ((D_801B9E5C != 1) || !(D_801B9E60 >= -(distResult - 50.0f))) {
```

Comment:
```suggestion
            if (!((D_801B9E5C == 1) && (D_801B9E60 >= -(distResult - 50.0f)))) {
```
Similar comment to the one above.

Reply:
Bump

Reply:
Sorry, somehow missed this review!